libsoftwarecontainer/src/gateway/cgroups/cgroupsparser.cpp

```diff
@@ -105,6 +105,36 @@ void CGroupsParser::parseCGroupsGatewayConfiguration(const json_t *element)
                 return;
             }
         }
+
+    } else if (("cpu.shares" == settingKey)) {
+        long int newValue;
+        try {
+            newValue = std::stol(settingValue);
+        } catch (std::out_of_range &err) {
+            std::string errorMessage = "The value for cpu.shares is too high";
+            log_debug() << errorMessage;
+            throw InvalidValueError(errorMessage);
+        } catch (std::invalid_argument &err) {
+            std::string errorMessage = "The value for cpu.shares is not an integer";
+            log_debug() << errorMessage;
+            throw InvalidValueError(errorMessage);
+        }
+
+        if (newValue < 2) {
+            std::string errorMessage = "Value of cpu.shares must be at least 2";
+            log_debug() << errorMessage;
+            throw InvalidValueError(errorMessage);
+        }
+        // if the new value is smaller/equal than the old value, then we don't save the new value.
+        if (m_settings.count(settingKey) != 0) {
+            if (std::stol(m_settings[settingKey]) >= newValue) {
```

Comment:
I can remove the debug printout, but not sure how you mean to change it to check that newValue is higher than the one set. I need a return for the case when it is <= newValue, since that implies we won't reach the setting on the variable.